src/auxilio/variaveis.py

```diff
@@ -16,4 +16,28 @@
               {"Matemática e suas tecnologias": ["$"]}]
 
 
+# Nome em ordem das colunas dos arquivos csv SIMUENEM
+nome_col_df_dados_alunos = ["Nome", "CPF", "2Lingua", "Curso", "Cota"]
+nome_col_df_gabarito = ["Área", "Questão", "Gabarito", "Matéria"]
+nome_col_df_respostas = ["Nome","CPF","2Lingua","Cota",
+                         "1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19",
+                         "20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37",
+                         "38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55",
+                         "56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73",
+                         "74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91",
+                         "92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108",
+                         "109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124",
+                         "125","126","127","128","129","130","131","132","133","134","135","136","137","138","139","140",
+                         "141","142","143","144","145","146","147","148","149","150","151","152","153","154","155","156",
+                         "157","158","159","160","161","162","163","164","165","166","167","168","169","170","171","172",
+                         "173","174","175","176","177","178","179","180"]
 
+# Códigos para 2lingua e posição respostas
+codigo_2lingua_dicionario = {"2Lingua":
+                                        {
+                                       #"2LinguaX": {"Código": 2, "Posição": [slice(x, y), slice(z,w)]}
+                                        "Inglês":   {"Código": 0, "Posição": [slice(0, 5), slice(10, 185)]},
```

Comment:
@matos1396, acho que aqueles valores em `slice(0, 5)` e `slice(10, 185)`, podem ser salvamos em variáveis, pois eles estão muito sem contexto, assim sendo [magics numbers (artigo explicando o que é isso)](https://stackoverflow.com/questions/47882/what-is-a-magic-number-and-why-is-it-bad). Pois me levanta o questionamente, por que começa em 0 e vai até 5, não em 1 e vai até 10? o mesmo para o 10 e 185. Talvez o ideal seja salvar o `slice` em si em uma variável dando assim mais contexto, porém acredito que essa ultima sugestão não seja tão necessária, o que você acha?

Reply:
@vianaz Eu concordo também, eu tive o mesmo pensamento em relação a essa questão do contexto deles, mas na hora foi a melhor forma que me veio de fazer. Não conhecia essa definição de Magic Numbers, vou dar uma pensada melhor em como melhorar isso.

Mas de forma geral, a ideia desses `slice` é para tornar a implementação das funções de correção para cada tipo de simulado mais genérica e conseguir usar a mesma função no futuro para os outros tipos de simulado, alterando/adicionando apenas os valores desses `slices` para cada lingua em cada tipo de simulado.

No `variaveis.py` eu poderia fazer algo do tipo:

```
# ENEM
posicao_enem_gabarito_questoes_ingles = [slice(0, 5), slice(10, 185)]
posicao_enem_gabarito_questoes_espanhol = [slice(5, 185), slice(None, None)]

# UFSC
posicao_ufsc_gabarito_questoes_2LinguaX = [slice(x,y), slice(w,z)]
posicao_ufsc_gabarito_questoes_2LinguaY = [slice(x,y), slice(w,z)]
posicao_ufsc_gabarito_questoes_2LinguaZ = [slice(x,y), slice(w,z)]
```

E no `codigo_2lingua_dicionario` a ideia seria adicionar um novo nível futuramente dessa forma:

```
codigo_2lingua_dicionario = {"2Lingua":
                                        {
                                        "Enem":
                                                {
                                                "Inglês":   {"Código": 0, "Posição": posicao_enem_gabarito_questoes_ingles},
                                                "Espanhol": {"Código": 1, "Posição": posicao_enem_gabarito_questoes_espanhol},
                                                "Quantidade": 5
                                                },
                                        "UFSC":
                                                {
                                                "2LinguaX": {"Código": 0, "Posição": posicao_ufsc_gabarito_questoes_2LinguaX},
                                                "2LinguaY": {"Código": 1, "Posição": posicao_ufsc_gabarito_questoes_2LinguaY},
                                                "2LinguaZ": {"Código": 2, "Posição": posicao_ufsc_gabarito_questoes_2LinguaZ}
                                                "Quantidade": 5
                                                }
                                        }
                            }
```

Dessa forma, no futuro caso precisasse alterar o número de questões, ou então se a ordem das questões mudasse nos inputs, bastaria apenas alterar os valores correspondentes nesse arquivo e a função de correção não precisaria ser alterada para voltar a funcionar.

Como você acha que ficaria melhor ?

Reply:
@matos1396 não ficou muito claro pra mim a ideia daquelas variáveis e nem muito a sua ideia. Porque vou pegar como exemplo aquelas variáveis que você sugeriu:

```python
# ENEM
posicao_enem_gabarito_questoes_ingles = [slice(0, 5), slice(10, 185)]
posicao_enem_gabarito_questoes_espanhol = [slice(5, 185), slice(None, None)]
```

Continuo com a dúvida de porque `posicao_enem_gabarito_questoes_ingles` o segundo `slice` tem valores e na variável `posicao_enem_gabarito_questoes_espanhol` o `slice` usa `None` como valor, o ideal é que ao qualquer pessoa bater o olho já fique claro a razão da utilização desses valores, não vejo que seja um problema de implementação (maneira que você faz ou planejou pro futuro), mas talvez de quebrar em mais variáveis ou talvez melhores nomes de variáveis, faz sentido?

Reply:
Entendi sim e concordo, vou pensar numa solução mais adequada e atualizo aqui em seguida

Reply:
@vianaz Oq você acha se no `variaveis.py` ficasse dessa forma:

```python
posicao = {
            "Inglês":   {"Início": 0, "Fim": 5}, 
            "Espanhol": {"Início": 5, "Fim": 10}
            #"2LinguaZ": {"Início": 10,"Fim": 15}
          }

codigo_2lingua_dicionario = {"2Lingua":
                                        {
                                        "Inglês":   {"Código": 0, "Posição": [posicao["Inglês"]["Início"], posicao["Inglês"]["Fim"]]},
                                        "Espanhol": {"Código": 1, "Posição": [posicao["Espanhol"]["Início"], posicao["Espanhol"]["Fim"]]},
                                        #"2LinguaZ": {"Código": 2, "Posição": [posicao["2LinguaZ"]["Início"], posicao["2LinguaZ"]["Fim"]]},
                                        "Quantidade": 5
                                        }
                            }
```

No caso do dicionário `posicao`, para cada 2 Lingua, o "Início" representa aonde que no input do gabarito começa as questões referentes a essa lingua e o "Fim" seria aonde termina essas questões.

No caso, os números representariam o "index" de cada linha do Gabarito de input sendo que o "Início" está incluso e o "Fim" não (análogo a notação de slices do python)

Exemplos:
`"Inglês":   {"Início": 0, "Fim": 5}` Representariam as linhas cujos index são: `0, 1, 2, 3, 4`
`"Espanhol":   {"Início": 5, "Fim": 10}` Representariam as linhas cujos index são: `5, 6, 7, 8, 9`

Dessa forma faria mais sentido para um leitor qualquer entender a razão e o significado dos números ai contidos.

No dicionário `codigo_2lingua_dicionario` nos valores para a "Posição" seria simplesmente uma lista com 2 números cada, o "Início" e o "Fim"

Então apenas para facilitar a visualização os valores correspondentes desse exemplo seriam assim:

```python
posicao = {
            "Inglês":   {"Início": 0, "Fim": 5}, 
            "Espanhol": {"Início": 5, "Fim": 10}
          }

codigo_2lingua_dicionario = {"2Lingua":
                                        {
                                        "Inglês":   {"Código": 0, "Posição": [0, 5]},
                                        "Espanhol": {"Código": 1, "Posição": [5, 10]},
                                        "Quantidade": 5
                                        }
                            }
```
 
Eu também poderia simplesmente implementar a correção sem a necessidade dessas variáveis relacionadas a "Posição", entretanto se futuramente a ordem das questões referentes a cada 2Lingua mudarem no arquivo do gabarito de input, teria que se alterar diretamente no código da correção.
A minha ideia era que se fosse feito dessa forma como proponho, só seria necessário alterar aqui as "Posições" no arquivo `variaveis.py` em vez de alterar o código da correção em si. O mesmo valeria para caso fosse adicionado uma terceira (ou mais) opções de "2Lingua".

Reply:
@matos1396 acredito que ter tudo nos `variaveis.py` é o que faz mais sentido! Pela mesma razão que você falou, em relação a sua implementação sugerida, está bem melhor agora, só mudaria o nome da variável `posicao` para `posicoes_linguas_na_planilha`.

Reply:
@vianaz Resolvido